[JENKINS-33557] - connectivity checks may cause setup wizard to hang for a long time

kzantow opened the pull request:
Issues with internet connectivity can manifest as very long wait times during Jenkins install. This reduces the wait time to 20 seconds and runs all (2) checks in parallel.

@jenkinsci/code-reviewers @reviewbybees esp. @tfennelly 

This addresses: https://issues.jenkins-ci.org/browse/JENKINS-33557

kzantow committed:
JENKINS-33557 - tcp timeouts cause installer to hang for 6 minutes

reviewbybees commented:
This pull request originates from a [CloudBees](https://www.cloudbees.com/) employee. At CloudBees, we require that all pull requests be reviewed by other CloudBees employees before we seek to have the change accepted. If you want to learn more about our process please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees).

kzantow committed:
Fix test for connectivity check

kzantow committed:
Merge remote-tracking branch 'primary/2.0' into JENKINS-33557-tcp-timeouts-during-install

kzantow committed:
Modify default ProxyConfiguration.open() to use 20 second connect
timeout, this may be modified with the configuration property:
jenkins.proxy.default_connect_timeout_millis, a value of 0 will restore
the previous behavior

kzantow commented on core/src/main/java/hudson/ProxyConfiguration.java:
```diff
@@ -218,22 +224,29 @@ public static ProxyConfiguration load() throws IOException {
+        if(p==null) {
+            con = url.openConnection();
+        } else {
+            con = url.openConnection(p.createProxy(url.getHost()));
```
Everything in this else block is just indentation

daniel-beck commented on core/src/main/java/hudson/ProxyConfiguration.java:
```diff
@@ -74,6 +74,12 @@
+     * Holds a default TCP connect timeout set on all connections returned from this class,
+     * note this is value is in milliseconds, it's passed directly to {@link URLConnection#setConnectTimeout(int)}
+     */
+    private static final int DEFAULT_CONNECT_TIMEOUT_MILLIS = Integer.getInteger("jenkins.proxy.default_connect_timeout_millis", 20 * 1000);
```
Naming inconsistent with all other System properties we have: https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties

It's camelCase except in the case of a STATIC_FINAL_FIELD_NAME.

kzantow committed:
Use standard property naming convention

daniel-beck commented on core/src/main/java/hudson/model/UpdateCenter.java:
```diff
@@ -1132,7 +1134,18 @@ public String getPluginRepositoryBaseUrl() {
             try {
-                Util.copyStreamAndClose(ProxyConfiguration.open(url).getInputStream(),new NullOutputStream());
+                URLConnection connection = (URLConnection) ProxyConfiguration.open(url);
+                connection.setConnectTimeout(20000); // JENKINS-33557 lower connection test timeout (20 sec)
```
Is this fixed to 20 seconds deliberately? Why not use reuse the default from `ProxyConfiguration`?

kzantow replied:
Ah, hadn't added the `ProxyConfiguration` value at that point, whoops - good catch.

kzantow committed:
Use ProxyConfiguration defaults in UpdateCenter
